lib/drawers/publish.vue

```diff
@@ -413,6 +442,24 @@
       },
       updateTime(val) {
         this.timeValue = val;
+      },
+      archivePage(archived) {
+        this.$store.dispatch('startProgress');
+        return this.$store.dispatch('updatePageList', { archived })
+          .then(() => {
+            this.$store.dispatch('finishProgress');
+            this.$store.dispatch('showSnackbar', {
+              message: `${archived ? 'Archived' : 'Unarchived'} page`,
+              action: 'Undo',
+              onActionClick: () => this.archivePage(!archived)
```

Comment:
out of curiosity what is this `onActionClick` function? where is it bound, etc?

Reply:
inside the snackbar component itself 😄